qiskit/providers/ibmq/job/ibmqjob.py

```diff
@@ -290,6 +290,45 @@ def status(self) -> JobStatus:
 
         return self._status
 
+    def done(self) -> bool:
+        """Return whether the job is done.
+
+        Returns:
+            True if job status is done, else false.
+        """
+        return self._is_job_status(JobStatus.DONE)
+
+    def running(self) -> bool:
+        """Return whether the job is running.
+
+        Returns:
+            True if job status is running, else false.
+        """
+        return self._is_job_status(JobStatus.RUNNING)
+
+    def cancelled(self) -> bool:
+        """Return whether the job is cancelled.
+
+        Returns:
+            True if job status is cancelled, else false.
+        """
+        return self._is_job_status(JobStatus.CANCELLED)
+
+    def _is_job_status(self, job_status: JobStatus) -> bool:
+        """Return whether the current job status matches the desired one.
+
+        Args:
+            job_status: the job status to check against.
+
+        Returns:
+            True if the current job status matches the desired one, else false.
+        """
+        if self._status in JOB_FINAL_STATES:
+            return self._status == job_status
+
+        self.status()  # Get the latest job status.
+        return self._status == job_status
```

Comment:
You don't actually need to check for `JOB_FINAL_STATES` since `job.status()` already does that. So this method can just be `return self.status() == job_status`